sapi/cli/php_cli_server.c

```diff
@@ -2353,7 +2353,81 @@ static void php_cli_server_client_dtor_wrapper(zval *zv) /* {{{ */
 	pefree(p, 1);
 } /* }}} */
 
-static int php_cli_server_ctor(php_cli_server *server, const char *addr, const char *document_root, const char *router) /* {{{ */
+/**
+ * Parse the host and port portions of an address specifier in
+ * one of the following forms:
+ * - hostOrIP
+ * - [hostOrIp]
```

Comment:
I wouldn't support host-only forms. If we get a host-only form, more likely than not someone just forgot to specify a port and isn't looking for an ephemeral port.

Reply:
I'm okay with that.